### DISCUSSION TITLE
Add a way to access `base` option from .astro component (such as `Astro.base`)

### DISCUSSION THREAD

**Discussion by kagankan** [withastro/roadmap#292] [Proposal]
## Use Case

an astro project with `base` config like this:

```js
export default defineConfig({
    base: '/site-base'
}
```

want to make a header component, placed in every page.
So these links should be root relative path in every page,
I want to get href attributes like this:

```html
<header>
    <a href="/site-base/">Home</a>
    <a href="/site-base/about/">About</a>
</header>
```

Then, `/site-base` should be determined by `base` config.
If there is `Astro.base` (returns `base` option), I can write like this:

```astro
<header>
    <a href={`${Astro.base}/`}>Home</a>
    <a href={`${Astro.base}/about/`}>About</a>
</header>
```

## Idea

How about adding `Astro.base`, which returns `base` option?

## Question

Can I make these links with current features?

**Comment by delucis** [comment]
Agreed. I actually expected `Astro.site` to be the combined values of the `site` and `base` config options (which would mean you could use `Astro.site.pathname` to get `base`), but doesn’t seem to be the case!

**Reply by Barnabas** [reply]
Yes, exposing `Astro.base` would be nice. If it did, another approach would be to use the `<base>` tag, like so:

```html
<head>
  <base href="{Astro.base}" />
</head>
<body>
  <header>
    <a href="./">Home</a>
    <a href="./about/">About</a>
```
If a base tag exists, then all relative links in the page use it instead of the current URL, which keeps your markup simple and static.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

**Reply by delucis** [reply]
There is an `import.meta.env.BASE_URL` environment variable, so maybe that is helpful here?

**Reply by kagankan** [reply]
I didn't know the `base` tag and the `import.meta.env.BASE_URL`!
These are very useful.

This is what I was looking for.

```js
export default defineConfig({
    base: '/site-base/'
}
```

```astro
<head>
  <base href={import.meta.env.BASE_URL} />
</head>
<body>
  <header>
    <a href="./">Home</a>
    <a href="./about/">About</a>
```

Thank you both!
